tests/unit_tests/template.py

```diff
@@ -0,0 +1,834 @@
+                         " AND EnterTime = 1").fetchone()
+    assert(reactor[0] == 1)
+
+    # check if 10 source facilities were deployed by NOInst
+    source = cur.execute("SELECT count(*) FROM agententry WHERE Prototype = 'source'"
+                         " AND EnterTime = 1").fetchone()
+    assert(source[0] == 1)
+
+    
+""" Test A_7 """
+reactor_source_growth = copy.deepcopy(template)
+reactor_source_growth['simulation'].update(
+   {   "region": {
+   "config": {"NullRegion": "\n      "}, 
+   "institution": [
+    {
+     "config": {
+      "NOInst": {
+       "calc_method": "arma", 
+       "demand_commod": "fuel_reactor", 
+       "demand_std_dev": "0.0", 
+       "growth_rate": "1.0", 
+       "initial_demand": "1", 
+       "prototypes": {"val": "source"}, 
+       "steps": "1", 
+       "supply_commod": "fuel"
+      }
+     }, 
+     "name": "source_inst"
+    }, 
+    {
+     "config": {
+      "NOInst": {
+       "calc_method": "arma", 
+       "demand_commod": "POWER", 
+       "demand_std_dev": "0.0", 
+       "growth_rate": "1.0", 
+       "initial_demand": "1", 
+       "prototypes": {"val": "reactor"}, 
+       "steps": "1", 
+       "supply_commod": "fuel_reactor"
+      }
+     },  
+     "name": "reactor_inst"
+    }
+   ], 
+   "name": "SingleRegion"
+  }}
+    )
+
+""" Test A_7 """
+def testA7_reactor_source_growth():
+    # tests if the reactor and source pair is correctly deployed with increase in demand
+    output_file = 'reactor_source_growth.sqlite'
+    input_file = output_file.replace('.sqlite','.json')
+    with open(input_file, 'w') as f:
+        json.dump(reactor_source_growth, f)
+    s = subprocess.check_output(['cyclus', '-o', output_file, input_file],
+                                universal_newlines=True, env=env)
+    # check if ran successfully
+    assert("Cyclus run successful!" in s)
+
+    # getting the sqlite file
+    cur = get_cursor(output_file)
+
+    # check if 1 reactor facility was deployed by NOInst at timestep 1
+    reactor = cur.execute("SELECT count(*) FROM agententry WHERE Prototype = 'reactor'"
+                         " AND EnterTime = 1").fetchone()
+    assert(reactor[0] == 1)
+
+    # check if 1 reactor facility was deployed by NOInst at timestep 13
+    reactor = cur.execute("SELECT count(*) FROM agententry WHERE Prototype = 'reactor'"
+                         " AND EnterTime = 13").fetchone()
+    assert(reactor[0] == 1)
+
+    # check if 1 source facility was deployed by NOInst at timestep 1
+    source = cur.execute("SELECT count(*) FROM agententry WHERE Prototype = 'source'"
+                         " AND EnterTime = 1").fetchone()
+    assert(source[0] == 1)
+
+    # check if 1 source facility was deployed by NOInst at timestep 13
+    source = cur.execute("SELECT count(*) FROM agententry WHERE Prototype = 'source'"
+                         " AND EnterTime = 13").fetchone()
+    assert(source[0] == 1)
+
+""" Test A_8 """
+"""
```

Comment:
This string seems to contain mostly Python code. Why?

Reply:
This was an old test that we forgot to get rid of. Sorry